tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -259,9 +259,14 @@ public bool AsBool(bool defaultValue, Func<bool, bool>? validator)
                 return value;
             }
 
-            // Horrible that we have to stringify the dictionary, but that's all that's available in the telemetry api
-            // _keys.Telemetry.Record(_keys.Key, string.Join(", ", defaultValue.Select(kvp => $"{kvp.Key}:{kvp.Value}")), ConfigurationOrigins.Default);
-            // return defaultValue;
+            if (getDefaultValue != null)
+            {
+                // Horrible that we have to stringify the dictionary, but that's all that's available in the telemetry api
+                var defaultValue = getDefaultValue();
+                Telemetry.Record(Key, string.Join(", ", defaultValue.Select(kvp => $"{kvp.Key}:{kvp.Value}")), true, ConfigurationOrigins.Default);
```

Comment:
Not related to this PR, but thisnaïve concatenation looks fishy. What if dictionary values have `,` or `:` or even `"` in them? (rhetorical)

Reply:
Right. But our parsing is equally naïve so the value would be rejected before getting there.

Reply:
hah, ok, thanks